### ISSUE TITLE
Queue full error when using routed_sender

### ISSUE THREAD

**Issue by jeffmontagna** [terascope/kafka-assets#484]
Running a job using the kafka_sender with the hash_router and routed_sender generates the error `Error: Local: Queue full`.  
- The job was using the id_reader to read an index from ES and write to 4 kafka clusters. 
- Tried all versions of Kafka 3.x using routed_sender with the same results. 
- Using a different kafka cluster made no difference
- Changing starting_key_depth made no difference
- Changing slice size made no difference
- Tried old style asset and new bundle type with no change. 
- Ran another job using datagenerator with  hash_router and routed_sender, and it did work without any errors.  
- Using the same job parameters, but changing the asset to `v2.8.2` and using kafka_sender with connection_map works without issue. 
 
Error from teraslice logs:
```
22:25:27.937Z ERROR teraslice: (assignment=worker, module=worker_context, worker_id=xxxx, ex_id=xxxx job_id=xxxx, err.code=-184)
    A slice error occurred {
      slice: {
        slice_id: 'xxxx',
        slicer_id: 0,
        slicer_order: 21,
        request: { key: 'abcd*', count: 6998 },
        _created: '2021-05-26T22:23:59.572Z'
      }
    }
    --
    Error: Local: Queue full
        at Function.createLibrdkafkaError [as create] (/app/node_modules/node-rdkafka/lib/error.js:423:10)
        at Producer.Client._errorWrap (/app/node_modules/node-rdkafka/lib/client.js:481:29)
        at Producer.produce (/app/node_modules/node-rdkafka/lib/producer.js:138:15)
        at ProducerClient.produce (/app/assets/9d89bd99e6fd09d8631cb4d9a325f6d93e3c0221/dist/_kafka_clients/producer-client.js:61:30)
        at KafkaSender.send (/app/assets/9d89bd99e6fd09d8631cb4d9a325f6d93e3c0221/dist/kafka_sender_api/sender.js:39:29)
        at concurrency (/app/assets/bb9dc2073ec283631361b442adcb169311e6d261/index.js:259172:21)
        at /app/assets/bb9dc2073ec283631361b442adcb169311e6d261/index.js:1753:37
```

**Comment by peterdemartini** [comment]
Is it possible we need to consider the batch of messages not be per route but per connection

**Comment by godber** [comment]
> Is it possible we need to consider the batch of messages not be per route but per connection

Who is this question directed to?

**Comment by peterdemartini** [comment]
Sorry that question was directed to the developer who fixes this issue

**Comment by ciorg** [comment]
fixed with mr #504
